### PR TITLE
awardee gets awarded xp when awardee is specified in the awardee_id field

### LINKED ISSUE #10: user_id might not always be the guy getting the xp
- figure it out
- efficiently
- and fast

### PR DESCRIPTION
**Pull request by cmfcmf**
Fixes #10.

### CODE REVIEW

File: config/gamification.yml [hunk 1]
```diff
@@ -14,6 +14,7 @@ events:
             - xp:
                 name: postXP
                 amount: 1
+                awardee_id: poster_id
```
Comment (cmfcmf) [inline config/gamification.yml]: Whoops, committed this by mistake.

**Comment by KBorchar** [comment]
We usually award XP to the user specified in the event's user_id field. This is implicit and not written into the yaml. This feature allows a different user to get XP, instead.

**Comment by corinnaj** [comment]
If I define another user inside the yaml, but send an event that does not have this field set, the original user will get the XP. Is that meant to happen? Shouldn't we get an error message that the payload isn't complete?

**Comment by KBorchar** [comment]
Did you test this? My understanding is, that nobody would get any XP in such a case; user_id would be set to nil in lines 16/17:

```        if (action['xp']['awardee_id']) {
          user_id = context.data.payload[action['xp']['awardee_id']];
```

**Comment by frederike-ramin** [comment]
@corinnaj , @KBorchar is right. the condition before the assignment handles if there is an assignee_id defined in the yaml, not the event. Nobody ever checks if it is defined in the event, so there should be an error when user_id is nil.

**Comment by corinnaj** [comment]
Sorry for the delay, I did test it, but now I feel like I must not have been on the correct branch? :sweat_smile: In any case, it's working for me now.

File: test/test-utils.js [hunk 2]
```diff
@@ -1,10 +1,11 @@
 const assert = require('assert');
 
 module.exports = {
-  async createEvent(app, user_id, event_name) {
+  async createEvent(app, user_id, event_name, payload = {}) {
```
Comment (corinnaj) [inline test/test-utils.js]: If we have this, do we still need the "createEventWithPayload" method?

**Pull request merged**